src/game_player.cpp

```diff
@@ -464,12 +464,14 @@ bool Game_Player::GetOnVehicle() {
 	data()->preboard_move_speed = GetMoveSpeed();
 	if (type != Game_Vehicle::Airship) {
 		data()->boarding = true;
-		if (!GetThrough()) {
-			SetThrough(true);
-			MoveForward();
-			SetThrough(false);
-		} else {
-			MoveForward();
+		if (!IsMoving() && !IsJumping()) {
```

Comment:
> Fix ship disembark behavior 

Disembark onto water? How is this possible? Is that a bug, if yes could you add a comment somewhere.

Reply:
Its in the test cases in #1429 

The code doesn't have anything specific to the "disembark into the water" bug.  It's just that it doesn't move again if you're already moving.

Do you still want some bug comment about that?

Reply:
no is fine, thanks for the explanation